lib/compiler.js

```diff
@@ -606,21 +607,51 @@ function resolveAppScriptPath(rootDir, relativePath, resolveOptions) {
   return (fixedFile === undefined ? resolvedPath : fixedFile);
 }
 
-function buildAllMixinInstructions(appRootDir, mixinDirs, options) {
+function buildAllMixinInstructions(appRootDir, mixinDirs, mixinSources, options,
+  modelInstructions) {
   var extensions = _.without(_.keys(require.extensions),
     _.keys(getExcludedExtensions()));
+
   var files = options.mixins || [];
+  var mixins = loadMixins(appRootDir, mixinDirs, files, extensions, options);
+  if (mixins === undefined) return;
```

Comment:
I find this difficult to grasp at the first sight, it looks like `loadMixins` should be modifying the `files` variable.